Libraries/LibWeb/Editing/Internal/Algorithms.cpp

```diff
@@ -1566,4 +1692,194 @@ void split_the_parent_of_nodes(Vector<GC::Ref<DOM::Node>> const& nodes)
         remove_extraneous_line_breaks_at_the_end_of_node(*last_node->parent());
 }
 
+// https://w3c.github.io/editing/docs/execCommand/#wrap
+GC::Ptr<DOM::Node> wrap(
+    Vector<GC::Ref<DOM::Node>> node_list,
+    Function<bool(GC::Ref<DOM::Node>)> sibling_criteria,
+    Function<GC::Ptr<DOM::Node>()> new_parent_instructions)
+{
+    // If not provided, sibling criteria returns false and new parent instructions returns null.
+    if (!sibling_criteria)
+        sibling_criteria = [](auto) { return false; };
+    if (!new_parent_instructions)
+        new_parent_instructions = [] { return nullptr; };
+
+    // 1. If every member of node list is invisible, and none is a br, return null and abort these steps.
+    auto any_node_visible_or_br = false;
+    for (auto& node : node_list) {
+        if (is_visible_node(node) || is<HTML::HTMLBRElement>(*node)) {
+            any_node_visible_or_br = true;
+            break;
+        }
+    }
+    if (!any_node_visible_or_br)
+        return {};
+
+    // 2. If node list's first member's parent is null, return null and abort these steps.
+    if (!node_list.first()->parent())
```

Comment:
Is `node_list` guaranteed to not be empty at this point?

Reply:
It is by virtue of all call sites always sending over at least one node, but that's a weak guarantee. I've added a `VERIFY`.